test/data/test_metrics.py

```diff
@@ -0,0 +1,33 @@
+from torchtext.data import metrics
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+class TestUtils(TorchtextTestCase):
+
+    def test_bleu_score(self):
+        # Full match
+        candidate = [['My', 'full', 'pytorch', 'test']]
+        refs = [[['My', 'full', 'pytorch', 'test'], ['Completely', 'Different']]]
+        assert metrics.bleu_score(candidate, refs) == 1
+
+        # No 4-gram
+        candidate = [['My', 'full', 'pytorch']]
+        refs = [[['My', 'full', 'pytorch', 'test'], ['Completely', 'Different']]]
+        assert metrics.bleu_score(candidate, refs) == 0
+
+        # Partial match
+        candidate = [['My', 'full', 'pytorch', 'test']]
+        refs = [[['My', 'full', 'pytorch', 'test', '!'], ['Different']]]
+        assert round(metrics.bleu_score(candidate, refs), 4) == 0.7788
```

Comment:
What's the source for these scores? Or how were they computed?

Reply:
I computed them "by hand" by applying the math mentioned in the paper. Should I make this more explicit?

Reply:
That's fine with me. Do you have more than 4 digits in your calculations? 

By the way, you could use "close to 0.7788" instead of round and equal.

Reply:
or `torch.testing.assert_allclose`